site/app/templates/officeHoursQueue/QueueStatus.twig

```diff
@@ -190,8 +201,86 @@
   </div>
 {% endif %}
 <script>
+
+    if(localStorage.getItem('QueueMessage')){
+        let current_time = new Date();
+        const sent = JSON.parse(localStorage.getItem('QueueMessageExpire'));
+        const queue = sent.queue;
+        const sent_time = sent.time;
+        const course = sent.course;
+        //get rid of message after 80 minutes
+        if(current_time.getTime() - 5000000 < sent_time) {
+            if(course === '{{ viewer.getCourse() }}'){
+                $('.queue-message').css('display', 'block');
+                $('#queue-message-msg').html(localStorage.getItem('QueueMessage'));
+                $('#queue-message-msg-time').html(sent.formatted_time);
+            }
+        } else {
+            localStorage.removeItem('QueueMessageExpire');
+            localStorage.removeItem('QueueMessage');
+        }
+
+    }
+
+     {% if viewer.inQueue() %}
+        function queueMessage(message, queue, course) {
+            if (queue === "{{viewer.getCurrentQueueCode()}}"){
+                let now = new Date();
+                const item = {
+                    queue: queue,
+                    time: now.getTime(),
+                    formatted_time: now.toLocaleString(),
+                    course: course
+                }
+                localStorage.setItem('QueueMessageExpire', JSON.stringify(item));
+                localStorage.setItem('QueueMessage', message);
+                $('.queue-message').css('display', 'block');
+                $('#queue-message-msg').html(message);
+                $('#queue-message-msg-time').html(now.toLocaleString());
+                if (Notification.permission === "granted") {
+                    // If it's okay let's create a notification
+                    let notification = new Notification("New Message From Office Hours Queue Staff");
+                }
+                // Otherwise, we need to ask the user for permission
+                else if (Notification.permission !== 'denied' || Notification.permission === "default") {
+                    Notification.requestPermission(function (permission) {
+                        if (permission === "granted") {
+                            let notification = new Notification("New Message From Office Hours Queue Staff");
+                        }
+                    });
+                //if notification permissions are denied
+                }else{
+                    alert('New Message From Office Hours Queue Staff');
+                }
+
+            }
+        }
+    {% endif %}
+
+    initSocketClientMsg();
+
+    function initSocketClientMsg() {
+        window.socketClient = new WebSocketClient();
+        window.socketClient.onmessage = (msg) => {
+            switch (msg.type) {
+                case "queue_message":
+                    queueMessage(msg.message, msg.queue, msg.course);
+                    break;
+            }
+        };
+        window.socketClient.open('office_hours_queue');
+    }
```

Comment:
We already have a websocket for this page. Could you add this type of message to that current one? Having 2 sockets for the same page seems unnecessary.